### PR TITLE
Make ZMCallStateTranscoder a request strategy

### PR DESCRIPTION
**Pull request by typfel**
For the ongoing project of removing the old state machine we need to convert all transcoder into request strategies.

### CODE REVIEW

File: Source/Synchronization/ZMCallStateTranscoder.h [hunk 1]
```diff
@@ -33,11 +34,11 @@ typedef NS_ENUM(uint8_t, ZMCallEventSource) {
 
 
 
-@interface ZMCallStateTranscoder : ZMObjectSyncStrategy <ZMObjectStrategy>
+@interface ZMCallStateTranscoder : ZMAbstractRequestStrategy <ZMEventConsumer, ZMContextChangeTrackerSource>
```
Comment (mikeger) [inline Source/Synchronization/ZMCallStateTranscoder.h]: May be would make sense then to rename the `ZMCallStateTranscoder` to `ZMCallStateRequestStrategy` ?

File: Tests/Source/Synchronization/Transcoders/ZMCallStateTranscoderTests.h [hunk 2]
```diff
@@ -21,14 +21,15 @@
 @import ZMCDataModel;
```
Comment (mikeger) [inline Tests/Source/Synchronization/Transcoders/ZMCallStateTranscoderTests.h]: I guess having the header is not necessary for the test class, unless if something is inheriting from this test case?

File: Source/Synchronization/ZMCallStateTranscoder.m [hunk 3]
```diff
@@ -101,22 +101,22 @@ - (instancetype)initWithManagedObjectContext:(NSManagedObjectContext *)moc
 }
 _Pragma("clang diagnostic pop")
```
Comment (mikeger) [inline Source/Synchronization/ZMCallStateTranscoder.m]: Is this dance with init still necessary?

Comment (typfel) [inline Source/Synchronization/ZMCallStateTranscoder.m]: No I just missed it. We should delete the init

File: Tests/Source/Synchronization/Transcoders/ZMCallStateTranscoderTests.h (hunk 2 above)
Comment (typfel) [inline Tests/Source/Synchronization/Transcoders/ZMCallStateTranscoderTests.h]: There's a an extension on the test

Comment (typfel) [inline Tests/Source/Synchronization/Transcoders/ZMCallStateTranscoderTests.h]: `ZMCallStateTranscoderTests (VideoCalling)`

**Comment by typfel** [comment]
@mikeger updated

**Pull request merged**